mkosi/gentoo.py

```diff
@@ -0,0 +1,553 @@
+            args_profile = "nomultilib"
+            # 20210711T170538Z/stage3-amd64-nomultilib-systemd-20210711T170538Z.tar.xz 214470580
+            regexp = f"^[0-9TZ]+/stage3-{self.arch}-{args_profile}-systemd-[0-9TZ]+[.]tar[.]xz"
+            all_lines = r.readlines()
+            for line in all_lines:
+                m = re.match(regexp, line.decode("utf-8"))
+                if m:
+                    stage3_tar = Path(m.group(0))
+                    break
+            else:
+                die("profile names changed upstream?")
+
+        stage3_url_path = urllib.parse.urljoin(
+            self.portage_cfg["GENTOO_MIRRORS"],
+            f"releases/{self.arch}/autobuilds/{stage3_tar}",
+        )
+        stage3_tar_path = self.portage_cfg["DISTDIR"] / stage3_tar
+        if not stage3_tar_path.is_file():
+            MkosiPrinter.print_step(f"Fetching {stage3_url_path}")
+            stage3_tar_path.parent.mkdir(parents=True, exist_ok=True)
+            urllib.request.urlretrieve(stage3_url_path, stage3_tar_path)
+
+        with tarfile.open(stage3_tar_path) as tfd:
+            MkosiPrinter.print_step(f"Extracting {stage3_tar.name}")
+            # GENTOO: usrmerge mess
+            bins = []
+            usr = []
+            usrreg = "^[.]/(etc|usr|var)"
+            binreg = "^[.]/(sbin|bin|lib|lib64)"
+            devreg = "^[.]/dev"
+            runreg = "^[.]/run"
+
+            members = tfd.getmembers()
+            usrpath = root / "usr"
+
+            ignore = ["./var/run", "./var/lock", "./bin/awk"]
+            for dir in self.UNINSTALL_IGNORE:
+                ignore += [dir, f"./usr/{dir}"]
+
+            for tarinfo in members:
+                if tarinfo.name in ignore or re.match(devreg, tarinfo.name) or re.match(runreg, tarinfo.name):
+                    if "build-script" in ARG_DEBUG:
+                        print(f"{tarinfo.name} -> /dev/null")
+                    continue
+                elif (
+                    tarinfo.issym()
+                    and (os.path.dirname(tarinfo.name) == "./usr/bin")
+                    and (os.path.dirname(tarinfo.linkname) == "../../bin")
+                ):
+                    if "build-script" in ARG_DEBUG:
+                        print(f"{tarinfo.name} -> {tarinfo.linkname} -> /dev/null")
+                    continue
+                elif re.match(usrreg, tarinfo.name):
+                    usr.append(tarinfo)
+                elif re.match(binreg, tarinfo.name):
+                    bins.append(tarinfo)
+                else:
+                    print(f"{tarinfo.name} -> /dev/null")
+                    continue
+
+            try:
+                tfd.extractall(usrpath, bins, numeric_owner=True)
+                tfd.extractall(root, usr, numeric_owner=True)
+            except FileExistsError:
+                pass
+
+            copy_path(root.joinpath("usr/sbin"), root.joinpath("usr/bin"))
+
+        return
```

Comment:
You can leave out the empty return, it makes no difference. Same for the other methods.